Pocket Casts Watch App Extension/WatchSyncManager.swift

```diff
@@ -164,12 +172,15 @@ class WatchSyncManager {
             let password = response[WatchConstants.Messages.LoginDetailsResponse.password] as? String ?? ""
             ServerSettings.setSyncingEmail(email: username)
             ServerSettings.saveSyncingPassword(password)
-            if SyncManager.isUserLoggedIn(), username.count == 0, password.count == 0 {
+
+            ServerSettings.appleAuthIdentityToken = response[WatchConstants.Messages.LoginDetailsResponse.appleAuthToken] as? String
+
+            if SyncManager.isUserLoggedIn(), username.isEmpty {
```

Comment:
Yeah, the username will be set on both Password and SSO flows. So if a user logs in with SSO, they'll have the username set but not a password. If there is no username, then we'll sign out. If we decide to try one of the other routes and fail to find the necessary Auth credentials, then it will throw an error and sign out as well, preserving the original flow.